### ISSUE TITLE
[BUG] ARM template appears to be modified erroneously by the SDK

### ISSUE THREAD

**Issue by SukruthKS** [Azure/azure-sdk-for-java#10141]
**Describe the bug**

One of the parameters provided in the ARM deployment template appears to be modified incorrectly by the Azure SDK. I'm upgrading the SDK in our system from version `1.18.0` to `1.32.1` and it was working fine on version `1.18.0`.

Parameter in the template before submitting deployment request:

```
{
    "properties": {
        "template": {
            "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
            "contentVersion": "1.0.0.0",
            "parameters": {
                ...
                ...
                "sfcUserAssignedIdentities": {
                    "type": "object",
                    "defaultValue": {
                        "/subscriptions/<redacted>/resourceGroups/<redacted>/providers/Microsoft.ManagedIdentity/userAssignedIdentities/<redacted>": {}
                        }
                    }
                },
```
Parameter after submitting the deployment request. The resource Id is broken down into a json fragment. I captured this by adding a logging interceptor on the client side and also observed the same through deployment history on Azure portal.
```
{
    "properties": {
        "template": {
            "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
            "contentVersion": "1.0.0.0",
            "parameters": {
                ...
                ...
                "sfcUserAssignedIdentities": {
                    "type": "object",
                    "defaultValue": {
                        "/subscriptions/<redacted>/resourceGroups/<redacted>/providers/Microsoft": {
                            "ManagedIdentity/userAssignedIdentities/<redacted>": {}
                        }
                    }
                },
```

Code for submitting deployment request:

```
azure.deployments().define(deplName)
     .withExistingResourceGroup(rgName)
     .withTemplate(templateJson)
     .withParameters("{}")
     .withMode(DeploymentMode.INCREMENTAL)
     .beginCreate();
```

***Exception or Stack Trace***
NA

**To Reproduce**
Provided in the description

***Code Snippet***
Provided in the description

**Expected behavior**
The SDK should not modify the ARM template in an inconsistent way.

**Screenshots**
NA

**Setup (please complete the following information):**
 - OS: Linux
 - IDE : IntelliJ
 - Version of the Library used: 1.32.1

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by SukruthKS** [comment]
From debugging a bit, this issue starts from version Azure SDK version `1.23.0` and happens due to [FlatteningSerializer](https://github.com/Azure/autorest-clientruntime-for-java/blob/master/client-runtime/src/main/java/com/microsoft/rest/serializer/FlatteningSerializer.java) which breaks up any json property whose name contains a `.`.

**Comment by weidongxu-microsoft** [comment]
Looks like a "@JsonFlatten" break this key (interpreting it as "properties.nestedProperty". I will investigate it today.

**Comment by weidongxu-microsoft** [comment]
Caused by a bug in dependency https://github.com/Azure/autorest-clientruntime-for-java/issues/663

This might mitigate the issue. Still double checking with owner of the dependency.
```
ObjectMapper objectMapper = new ObjectMapper();

azure.deployments().define(deplName)
     .withExistingResourceGroup(rgName)
     .withTemplate(objectMapper.readValue(templateJson, new TypeReference<Map<String,Object>>(){}))
     .withParameters("{}")
     .withMode(DeploymentMode.INCREMENTAL)
     .beginCreate();
```

**Comment by sfc-gh-skomarla** [comment]
Thanks @weidongxu-microsoft! I will give it a shot tomorrow.

**Comment by SukruthKS** [comment]
Thanks @weidongxu-microsoft! I will give it a shot tomorrow.

**Comment by SukruthKS** [comment]
The above fix worked for us!

**Comment by weidongxu-microsoft** [comment]
@SukruthKS Good to know the workaround works.

I will continue work with owner of the dependency lib on the fix to root cause. If things work as expected, next release (1.34.0, scheduled on 1st May) should have it fixed.

**Comment by weidongxu-microsoft** [comment]
Dependency not fixed. Mitigated in SDK. To be released for 1.34.0, around 1st May.
